api/src/main/java/org/openmrs/module/fgh/mpi/SnapshotEventProcessor.java

```diff
@@ -128,14 +138,70 @@ public void process(DatabaseEvent event) {
+					int successPatientCount = 0;
+					
+					if (isOpenCrMPI(mpiSystem)) {
+						Map<String, Object> fhirBundle = new HashMap(3);
+						fhirBundle.put(MpiConstants.FIELD_RESOURCE_TYPE, MpiConstants.BUNDLE);
+						fhirBundle.put(MpiConstants.FIELD_TYPE, MpiConstants.BATCH);
+						fhirBundle.put(MpiConstants.FIELD_ENTRY, fhirPatients);
+						
+						List<Object> response = mpiHttpClient.submitBundle("fhir", mapper.writeValueAsString(fhirBundle),
+						    List.class);
+						
+						successPatientCount = response.size() / 2;
+					} else if (isSanteMPI(mpiSystem)) {
+						Map<String, Object> fhirMessageHeaderEntry = generateMessageHeader();
+						
+						//The entry of resource in message bundle message
+						Map<String, Object> fhirResourceEntry = new HashMap<>(2);
+						
+						fhirResourceEntry.put("fullUrl", "metadata.epts.e-saude.net/bundle");
+						fhirResourceEntry.put("resource", new HashMap<>(3));
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_RESOURCE_TYPE,
+						    MpiConstants.BUNDLE);
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_TYPE,
+						    MpiConstants.FIELD_TYPE_HISTORY);
+						
+						List<Map<String, Object>> fhirPatintsPlusRequest = new ArrayList<Map<String, Object>>(
+						        fhirPatients.size());
+						
+						for (Map<String, Object> fhirPatient : fhirPatients) {
+							Map<String, Object> resourceEntry = getObjectOnMapAsMap(MpiConstants.FIELD_RESOURCE,
+							    fhirPatient);
+							
+							Object patientUuid = getObjectOnMapAsListOfMap(MpiConstants.FIELD_IDENTIFIER, resourceEntry)
+							        .get(0).get("value");
+							
+							Map<String, Object> patientEntry = fastCreateMap(MpiConstants.FIELD_RESOURCE, resourceEntry,
+							    "request", fastCreateMap("method", "POST", "url", "Parient/" + patientUuid), "fullUrl",
+							    "Parient/" + patientUuid);
+							
+							fhirPatintsPlusRequest.add(patientEntry);
+						}
+						
+						getObjectOnMapAsMap("resource", fhirResourceEntry).put(MpiConstants.FIELD_ENTRY,
+						    fhirPatintsPlusRequest);
+						
+						Map<String, Object> messageBundle = new HashMap<String, Object>();
+						messageBundle.put("resourceType", "Bundle");
+						messageBundle.put("type", "message");
+						messageBundle.put(MpiConstants.FIELD_ENTRY, new ArrayList<Map<String, Object>>(2));
+						
+						getObjectOnMapAsListOfMap(MpiConstants.FIELD_ENTRY, messageBundle).add(fhirMessageHeaderEntry);
+						getObjectOnMapAsListOfMap(MpiConstants.FIELD_ENTRY, messageBundle).add(fhirResourceEntry);
+						
+						Map<String, Object> response = mpiHttpClient.submitBundle("fhir/Bundle",
+						    mapper.writeValueAsString(messageBundle), Map.class);
+						
+						//Sante doen't return the amount of pacients successifuly received but if there is any error in one patient Sante throws Exception
+						//Force successPatientCount to be equal fhirPatients.size()
```

Comment:
This is worrying me, what does sante MPI return when you submit a batch, do you mind sharing it?

Reply:
This is the response after successifuly submited a bundle of 3 patients

```
{
    "resourceType": "Bundle",
    "meta": {
        "lastUpdated": "2023-04-14T11:33:43.494784+00:00"
    },
    "type": "message",
    "timestamp": "2023-04-14T11:33:45.142946+00:00",
    "entry": [
        {
            "fullUrl": "urn:uuid:0592d61d-a6a3-4476-9c00-6c9deeec5fbc",
            "resource": {
                "resourceType": "MessageHeader",
                "id": "0592d61d-a6a3-4476-9c00-6c9deeec5fbc",
                "response": {
                    "code": "ok",
                    "details": {
                        "reference": "urn:uuid:d41c4bfe-1852-4614-b61d-e4e5a4d36689"
                    }
                }
            }
        },
        {
            "fullUrl": "urn:uuid:d41c4bfe-1852-4614-b61d-e4e5a4d36689",
            "resource": {
                "resourceType": "OperationOutcome",
                "id": "d41c4bfe-1852-4614-b61d-e4e5a4d36689",
                "issue": [
                    {
                        "severity": "information",
                        "diagnostics": "Register/Update Patient (K:5a8dd065-180b-4cee-bea0-11bb5920251a, V:0fb05be0-d078-4d85-8dba-a791d9f1b7fd)"
                    }
                ]
            }
        }
    ]
}
```